### PR TITLE
Add support for GIT_SYNC_MAX_FAILURES

### LINKED ISSUE #157: add support for GIT_SYNC_MAX_SYNC_FAILURES
Currently, we dont let the user set `GIT_SYNC_MAX_SYNC_FAILURES` in the `git-sync` container.

We should allow users to configure this feature with a value like `dags.gitSync.maxFailures`.

This feature may be useful for users as with flaky network connections, who dont want to rely on the exponential backoff provided by Kubernetes, as requested here: https://github.com/airflow-helm/charts/pull/118#issuecomment-822763842

### PR DESCRIPTION
**Pull request by n-oden**
**What issues does your PR fix?**

- fixes #157 

**What does your PR do?**

Add support for GIT_SYNC_MAX_FAILURES.

## Checklist

- [X] Commits are [signed](https://github.com/airflow-helm/charts/tree/main/CONTRIBUTING.md#sign-your-work)
- [X] Commits are [squashed](https://github.com/airflow-helm/charts/tree/main/CONTRIBUTING.md#squash-commits) (if appropriate)
- [X] Chart [version bumped](https://github.com/airflow-helm/charts/tree/main/CONTRIBUTING.md#versioning)
- [X] Chart [documentation updated](https://github.com/airflow-helm/charts/tree/main/CONTRIBUTING.md#documentation)
- [X] Passes [linting](https://github.com/airflow-helm/charts/tree/main/CONTRIBUTING.md#linting)

Signed-off-by: Nathan J. Mehl <n@oden.io>

### CODE REVIEW

**Comment by n-oden** [comment]
@thesuperzapper ping? :)

**Comment by n-oden** [comment]
@gsemet @thesuperzapper gentle poke? :)

**Comment by gsemet** [comment]
@thesuperzapper I would agree, do you ?

**Comment by n-oden** [comment]
Bumped again since 8.1.0 landed without this.  @gsemet @thesuperzapper should this be a patch release or a minor release?

**Comment by n-oden** [comment]
Bumped again to sit atop 8.1.3.  @gsemet @thesuperzapper is there anything else that needs to be done here?

File: charts/airflow/UPGRADE.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 # Upgrading Steps
 
+## `v8.1.3` → `v8.1.4`
```
Comment (thesuperzapper) [inline charts/airflow/UPGRADE.md:3]: Please fix after renaming version

File: charts/airflow/README.md [hunk 2]
```diff
@@ -59,6 +59,8 @@ kubectl port-forward --namespace $NAMESPACE $POD_NAME 8080:8080
 
 Old Version | New Version | Upgrade Guide
 --- | --- | ---
+v8.1.3 | v8.1.4 | [link](UPGRADE.md#v813x--v814)
+v8.1.1 | v8.1.2 | [link](UPGRADE.md#v811x--v812)
```
Comment (thesuperzapper) [inline charts/airflow/README.md:63]: Please fix after renaming version

File: charts/airflow/Chart.yaml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
 description: the community-maintained descendant of the stable/airflow helm chart - focused on ease-of-use
 name: airflow
-version: 8.1.3
+version: 8.1.4
```
Comment (thesuperzapper) [inline charts/airflow/Chart.yaml:4]: This change will need to be a minor patch (as we are adding a value)

In this case, we can do `8.3.0`

File: charts/airflow/values.yaml [hunk 4]
```diff
@@ -1080,6 +1080,10 @@ dags:
     ##
     sshKnownHosts: ""
 
+    ## the number of consecutive failures allowed before aborting
+    ## (the first sync must succeed, -1 will retry forever after the initial sync)
```
Comment (thesuperzapper) [inline charts/airflow/values.yaml:1084]: To be consistent with all other docstrings, this should be:
```
## the number of consecutive failures allowed before aborting
## 
## NOTE:
##  - the first sync must succeed, -1 will retry forever after the initial sync
##
```

**Review by thesuperzapper — CHANGES_REQUESTED** [review]
@n-oden lets merge this after we merge https://github.com/airflow-helm/charts/pull/200, which will create version `8.2.0`

**Comment by n-oden** [comment]
All sounds good to me; I'll update after #200 has landed

**Comment by thesuperzapper** [comment]
@n-oden `8.2.0` has been released with #200

**Comment by n-oden** [comment]
@thesuperzapper rebased and updated

**Comment by n-oden** [comment]
@thesuperzapper gentle nudge. :)

File: charts/airflow/UPGRADE.md [hunk 5]
```diff
@@ -1,10 +1,18 @@
 # Upgrading Steps
 
-## `v8.1.X` → `v8.2.0`
+<<<<<<< HEAD
```
Comment (thesuperzapper) [inline charts/airflow/UPGRADE.md:3]: @n-oden what is this line?

Comment (n-oden) [reply charts/airflow/UPGRADE.md:3]: arrrrrrrrrgh :(  Argh is what it is.

Comment (n-oden) [reply charts/airflow/UPGRADE.md]: and now gone is what it is :)

File: charts/airflow/values.yaml [hunk 6]
```diff
@@ -1080,6 +1080,13 @@ dags:
     ##
     sshKnownHosts: ""
 
+    ## the number of consecutive failures allowed before aborting
+    ##
+    ## NOTE:
+    ##  - the first sync must succeed
+    ##  - a value of -1 will retry forever after the initial sync
```
Comment (thesuperzapper) [inline charts/airflow/values.yaml:1087]: @n-oden I am super sorry to be tedious, but can you add the `##` at the bottom of the docstring.

For example,
```
    ## the number of consecutive failures allowed before aborting
    ##
    ## NOTE:
    ##  - the first sync must succeed
    ##  - a value of -1 will retry forever after the initial sync
    ##
    maxFailures: 0
```

Comment (n-oden) [reply charts/airflow/values.yaml:1087]: trust me I am the last person in the world to object to fussy formatting feedback 😂 

Updated!

**Review by thesuperzapper — APPROVED** [review]
@n-oden looks good, sorry for the extended back-and-forwards!

**Comment by n-oden** [comment]
@thesuperzapper no worries -- we're not even going to be able to personally use this until we're ready to upgrade to airflow 2.0, so I'm just happy it'll be waiting for us then. :)

**Pull request merged**